Validation/RecoTau/src/TauValidationMiniAOD.cc

```diff
@@ -157,28 +536,205 @@ void TauValidationMiniAOD::analyze(const edm::Event& iEvent, const edm::EventSet
+
+      // fill tau mass for decay modes 0,1+2,5,6,7,10,11
+      if (matchedTau->decayMode() == 0) {
+        mtau_dm0Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm0Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 1 || matchedTau->decayMode() == 2) {
+        mtau_dm1p2Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm1p2Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 5) {
+        mtau_dm5Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm5Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 6) {
+        mtau_dm6Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm6Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 10) {
+        mtau_dm10Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm10Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 11) {
+        mtau_dm11Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm11Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      }
+
+      // fill decay mode population plot
+      ntau_vs_dmMap.find("")->second->Fill(taus->size(), matchedTau->decayMode());
+
+      //Fill decay mode migration 2D histogragms
+      //First do a gen Matching
+      unsigned genindex = 0;
+      for (std::vector<reco::GenParticle>::const_iterator genParticle = genParticles->begin();
+           genParticle != genParticles->end();
+           genParticle++) {
+        if (abs(genParticle->pdgId()) == 15) {
+          float gendR = deltaR(matchedTau->eta(), matchedTau->phi(), genParticle->eta(), genParticle->phi());
+          if (gendR < gendRmin) {
+            gendRmin = gendR;
+            genmatchedTauIndex = genindex;
+          }
+        }
+        genindex = genindex + 1;
+      }
+
+      if (gendRmin < 0.15) {
+        int nPhotonsFromTauDecay = 0;
+        int nPi0s = 0;
+        int nPis = 0;
+        auto &gentau = genParticles->at(genmatchedTauIndex);
+        for (unsigned idtr = 0; idtr < gentau.numberOfDaughters(); idtr++) {
+          const reco::GenParticle *dtr = dynamic_cast<const reco::GenParticle *>(gentau.daughter(idtr));
+          int dtrpdgID = std::abs(dtr->pdgId());
+          int dtrstatus = dtr->status();
+          if (dtrpdgID == 12 || dtrpdgID == 14 || dtrpdgID == 16)
+            continue;
+          if (dtrpdgID == 111 || dtrpdgID == 311)
+            nPi0s++;
+          else if (dtrpdgID == 211 || dtrpdgID == 321)
+            nPis++;
+          else if (dtrpdgID == 22) {
+            float dr_taugamma = deltaR(gentau.eta(), gentau.phi(), dtr->eta(), dtr->phi());
```

Comment:
Using `deltaR2` is computationally cheaper (here and elsewhere in this code)

Reply:
Done